portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -55,24 +95,51 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       maxCommentsToDisplay = DEFAULT_MAX_COMMENTS;
     }
 
+    // Obtain the page ID of comment history to display.
+    String pageIdParam = request.getParameter("pageId");
+    totalPages = (int) Math.ceil(((double) totalComments) / maxCommentsToDisplay);
+    try {
+      currentPageId = Integer.parseInt(pageIdParam);
```

Comment:
I think there should be a limit on the number of pages like maxPages and checks for that. Right now I can say take me to page 100 and it will say "Currently on page 100"

Reply:
Yes you're right. I'm implementing that and will submit a code change soon

Reply:
This issue is fixed, along with some other missed edges. Thanks for pointing this out!